spark/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala

```diff
@@ -254,7 +254,15 @@ object Protocol {
 
     val (readerVersion, writerVersion, enabledFeatures) =
       minProtocolComponentsFromMetadata(spark, metadata)
-    Protocol(readerVersion, writerVersion).withFeatures(enabledFeatures)
+    // New table protocols should always be denormalized and then normalized to convert the
+    // protocol to the weakest possible form. For example:
+    // 1) (3, 7, RowIDs) is normalized to (3, 7, RowIDs).
```

Comment:
`RowId` -> `RowTracking`. Also "weakest possible form" is not clear to the casual reader. Can you define this and add an `i.e.` to the comment?